src/view/com/post-thread/PostThread.tsx

```diff
@@ -163,42 +169,81 @@ function PostThreadLoaded({
     [threadViewPrefs, thread],
   )
 
-  // construct content
+  const [readyToRender, setReadyToRender] = React.useState(false)
+  const [topPageCount, setTopPageCount] = React.useState(1)
+  const isPrepending = React.useRef(false)
+
   const posts = React.useMemo(() => {
-    let arr = Array.from(
-      flattenThreadSkeleton(
-        sortThread(thread, threadViewPrefs),
-        hasSession,
-        treeView,
-      ),
+    const items = createThreadSkeleton(
+      sortThread(thread, threadViewPrefs),
+      hasSession,
+      treeView,
+      readyToRender,
     )
+
+    const highlightedPost = isThreadPost(items.highlightedPost)
+      ? items.highlightedPost
+      : undefined
+
+    // Create the posts to prepend
+    let postsToPrepend = items.parents?.slice(-(topPageCount * 15)) ?? []
+    if (postsToPrepend.length > 0) {
+      // If there are any parent posts, we need to add the header here
+      postsToPrepend = [TOP_COMPONENT, ...postsToPrepend]
+    } else if (
+      postsToPrepend.length === 0 &&
+      !highlightedPost?.ctx.isParentLoading
+    ) {
+      // If there are no parents and there isn't a parent loading, then we just add the top component
+      postsToPrepend = [TOP_COMPONENT]
+    }
```

Comment:
We need to handle the logic of adding the view header here now, since we are continually loading in posts at the top.